crypto/BRCryptoKey.c

```diff
@@ -113,15 +126,34 @@ cryptoKeyFindAddressParams (const char *string) {
     return EMPTY_ADDRESS_PARAMS;
 }
 
+extern BRCryptoKey
+cryptoKeyCreateFromStringProtectedPrivate (const char *privateKey, const char * passphrase) {
+    if (!BRBIP38KeyIsValid (privateKey)) return NULL;
```

Comment:
So with this function we create a `BRCryptoKey` based on BIP38.  Once returned, all 'BIP38 context' is lost - meaning that `BITCOIN_ADDRESS_PARAMS`will apply in any subsequent `BRCryptoKey` uses.  Is that a problem?

Reply:
Based on iOS' usage, it does not appear to be.

They extend the `String` class with `isValidPrivateKey` and `isValidBip38Key` methods (the latter of which `cryptoKeyIsProtectedPrivate` is analogous to).

Beyond that, they have a `BRKey(bip38Key: address, passphrase: password)` method, which `cryptoKeyCreateFromStringProtectedPrivate (const char *privateKey, const char * passphrase)` is analogous to.

The address params associated with a key are used for serializing, at the moment. It doesn't look like that is used with the BIP38 logic. If that ends up being needed in the future, the serialization logic likely needs to be extended (i.e. support for specifying a password, maybe support for specifying a network, etc.).